common/match/matching.perf.ts

```diff
@@ -0,0 +1,339 @@
+            });
+        }
+    }
+
+    for (const offer of offers) {
+        const uuid = `${offer.studentId}`;
+        if (helperByUUID.has(uuid)) {
+            helperByUUID.get(uuid).matchRequestCount += 1;
+        } else {
+            offerByUUID.set(uuid, offer);
+            helperByUUID.set(uuid, {
+                id: offer.studentId,
+                uuid,
+                matchRequestCount: 1,
+                subjects: offer.subjects,
+                createdAt: offer.requestAt,
+                excludeMatchesWith: (matchesPerStudent[uuid] ?? []).map((uuid) => ({ uuid })),
+                state: offer.state,
+                // firstMatchRequest: student.firstMatchRequest
+            });
+        }
+    }
+
+    const result = match([...helperByUUID.values()], [...helpeeByUUID.values()], {
+        balancingCoefficients: {
+            subjectMatching: 0.65,
+            state: 0.05,
+            waitingTime: 0.2,
+            matchingPriority: 0.1,
+        },
+    });
+
+    const matching: Matching = result.matches.map((it) => ({
+        request: requestByUUID.get(it.helpee.uuid),
+        offer: offerByUUID.get(it.helper.uuid),
+    }));
+
+    return matching;
+}
+
+describe('Real World Matching Performance', () => {
+    test.each([
+        // New Algorithm
+        [
+            'new',
+            1000,
+            {
+                matchCountSum: 1041,
+                matchCountAvg: 520.5,
+                matchingSubjectsSum: 1954,
+                matchingSubjectsAvg: 1.877041306436119,
+                matchingState: 484,
+                pupilWaitingTimeAvg: 294.63748241129935,
+                studentWaitingTimeAvg: 328.5918544018619,
+                matchRuns: 2,
+            },
+        ],
+        [
+            'new',
+            10,
+            {
+                matchCountSum: 1044,
+                matchCountAvg: 15.582089552238806,
+                matchingSubjectsSum: 1793,
+                matchingSubjectsAvg: 1.7174329501915708,
+                matchingState: 332,
+                pupilWaitingTimeAvg: 20.26238745497861,
+                studentWaitingTimeAvg: 61.03486072712683,
+                matchRuns: 67,
+            },
+        ],
+        [
+            'new',
+            1,
+            {
+                matchCountSum: 1044,
+                matchCountAvg: 2.4110854503464205,
+                matchingSubjectsSum: 1795,
+                matchingSubjectsAvg: 1.7193486590038314,
+                matchingState: 323,
+                pupilWaitingTimeAvg: 17.05913367225194,
+                studentWaitingTimeAvg: 57.4907587186702,
+                matchRuns: 433,
+            },
+        ],
+        // Old Algorithm
+        // Interestingly the old algorithm is not deterministic
+        // Also running it in the barrier does not make sense as it wont change anyways
+        /* [
+            'old',
+            1000,
+            {
+                matchCountSum: 883,
```

Comment:
It's interesting that the old algorithm prioritizes the number of matching states over the number of matching subjects and the number of matches created ...